### PR TITLE
Feat/Automated e2e test framework

### PR DESCRIPTION
**Pull request by WaltId-MikeRichardson**
Phase I of the automated e2e test framework covering wallet and issuer apis. See README in the test directory for comments on architecture, running the tests etc.

### CODE REVIEW

**Review by mikeplotean — APPROVED** [review]
maybe yo can check the couple of comments
otherwise, looks good!

File: waltid-wallet-api/src/main/kotlin/id/walt/webwallet/db/Db.kt [hunk 1]
```diff
@@ -80,24 +70,44 @@ object Db {
...
-
+            
+            
             SchemaUtils.drop(*(tables.reversedArray()))
             SchemaUtils.create(*tables)
-
+            
             runBlocking {
+                
+                AccountsService.register(request = EmailAccountRequest("Max Mustermann", "string@string.string", "string"))
```
[the remark this answers is not in the document]
Comment (mikeplotean) [reply waltid-wallet-api/src/main/kotlin/id/walt/webwallet/db/Db.kt:83]: correction: adding the default issuer here is not required, as it is added inside the `register` method

[the remark this answers is not in the document]
Comment (WaltId-MikeRichardson) [reply waltid-wallet-api/src/main/kotlin/id/walt/webwallet/db/Db.kt:83]: I've taken default issuer out. I'm not sure we need it.

File: waltid-wallet-api/src/main/kotlin/id/walt/webwallet/db/IssuanceExamples.kt
[the remark this answers is not in the document]
Comment (WaltId-MikeRichardson) [reply waltid-wallet-api/src/main/kotlin/id/walt/webwallet/db/IssuanceExamples.kt:1]: Moved to utils folder

File: waltid-wallet-api/src/test/kotlin/Credential.kt
[the remark this answers is not in the document]
Comment (WaltId-MikeRichardson) [reply waltid-wallet-api/src/test/kotlin/Credential.kt:1]: Resused credential from utils, this file now deleted

File: waltid-wallet-api/src/test/kotlin/E2EWalletTestDeployed.kt [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                password,
+                "email"
+            )
+        )
+    }
+    
+    @Test
+    fun e2eTestAuthentication() = runTest {
+        deployedClient = newClient()
+//        testCreateUser(User(name="tester", email="tester@email.com", password="password", accountType="email"))
```
[the remark this answers is not in the document]
Comment (WaltId-MikeRichardson) [reply waltid-wallet-api/src/test/kotlin/E2EWalletTestDeployed.kt:57]: I commented it back in and removed the status ok check. This function creates a new user but the user might already be there so I commented it out temporarily, it's  back in and working correctly now.

File: waltid-wallet-api/src/test/kotlin/E2EWalletTestLocal.kt [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+            WalletConfigManager.preloadConfig(
+                "web", WalletWebConfig()
+            )
+            webWalletSetup()
+            WalletConfigManager.loadConfigs(emptyArray())
+        }
+    }
+
+    private fun ApplicationTestBuilder.newClient(token: String? = null) = createClient {
```
[the remark this answers is not in the document]
Comment (WaltId-MikeRichardson) [reply waltid-wallet-api/src/test/kotlin/E2EWalletTestLocal.kt:70]: I had a look at this but couldn't see any real value in creating an abstract function, especially as the local version sets
the function for `ApplicationTestBuilder `which doesn't seem to be compatible with a simple `httpClient`.

**Review by mikeplotean — DISMISSED** [review]

File: waltid-wallet-api/src/main/kotlin/id/walt/webwallet/service/SSIKit2WalletService.kt [hunk 4]
```diff
@@ -346,24 +333,15 @@ class SSIKit2WalletService(
...
-        install(ContentNegotiation) {
-            json()
-        }
-        install(Logging) {
-            logger = Logger.SIMPLE
-            level = LogLevel.ALL
-        }
-        followRedirects = false
-    }
+    val http = getHttpClient()
```
[the remark this answers is not in the document]
Comment (WaltId-MikeRichardson) [reply waltid-wallet-api/src/main/kotlin/id/walt/webwallet/service/SSIKit2WalletService.kt]: I had a look at this: good suggestion. `httpClient `was being initialized in both `WalletServiceManager `and `SSIKitWalletService`. 

Now I just do that in `WalletServiceManager `and pass the object in to the latter's constructor.

**Review by mikeplotean — APPROVED** [review]

**Pull request merged**